addons/crm/models/crm_lead.py

```diff
@@ -900,7 +900,7 @@ def _read_group_stage_ids(self, stages, domain, order):
         # - OR ('team_ids', '=', team_id), ('fold', '=', False) if team_id: add team columns that are not folded
```

Comment:
The changes don't seem to allign with the original approach of the method.

Assuming the original approach was wrong, we should update the comments.